File: bindings/csharp/GetOrderHistoryRequest.cs
namespace OpenLimits
{
    public struct GetOrderHistoryRequest
    {
        public readonly string market;
        public readonly Paginator paginator;

        public GetOrderHistoryRequest(string market, Paginator paginator)
        {
            this.market = market;
            this.paginator = paginator;
        }

        public GetOrderHistoryRequest(string market)
        {
            this.market = market;
            this.paginator = null;
        }
    }
}

File: crates/openlimits-exchange/src/model/request/get_order_history_request.rs
use derive_more::Constructor;
use serde::Deserialize;
use serde::Serialize;
use crate::model::{OrderStatus, Paginator};
use crate::model::market_pair::MarketPair;

/// This struct represents the historic of the orders
#[derive(Serialize, Deserialize, Clone, Constructor, Debug)]
pub struct GetOrderHistoryRequest {
    pub market_pair: Option<MarketPair>,
    pub order_status: Option<Vec<OrderStatus>>,
    pub paginator: Option<Paginator>,
}